content/md/en/docs/tutorials/smart-contracts/erc20-token.md

```diff
@@ -0,0 +1,652 @@
+
+To get started, lets create a new project with some template code.
+
+To build an ERC-20 token smart contract:
+
+1. Open a terminal shell on your local computer, if you don’t already have one open.
+
+1. Create a new project named `erc20` by running the following command:
+    
+    ```bash
+    cargo contract new erc20
+    ```
+
+1. Change to the new project directory by running the following command:
+    
+    ```bash
+    cd erc20/
+    ```
+
+1. Open the `lib.rs` file in a text editor.
+    
+1. Replace the default template source code with new [erc20](https://github.com/substrate-developer-hub/substrate-contracts-workshop/blob/master/2/assets/2.1-template.rs) source code.
+
+1. Save the changes to the `lib.rs` file, then close the file.
+
+1. Open the `Cargo.toml` file in a text editor and review the dependencies for the contract.
+
+1. In the `[dependencies]` section, modify the `scale` and `scale-info` settings, if necessary.
+    
+    ```toml
+    scale = { package = "parity-scale-codec", version = "3", default-features = false, features = ["derive"] }
+    scale-info = { version = "2", default-features = false, features = ["derive"], optional = true }
+    ```
+
+1. Save changes to the `Cargo.toml` file, then close the file.
+
+1. Verify that the program compiles and passes the trivial test by running the following command:
+    
+    ```bash
+    cargo +nightly test
+    ```
+
+    The command should display output similar to the following to indicate successful test completion:
+
+    ```text
+    running 2 tests
+    test erc20::tests::new_works ... ok
+    test erc20::tests::balance_works ... ok
+    
+    test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
+    ```
+
+1. Verify that you can build the WebAssembly for the contract by running the following command:
+    
+    ```bash
+    cargo +nightly contract build
+    ```
+    
+    If the program compiles successfully, you are ready to upload it in its current state or start adding functionality to the contract.
+
+## Upload and instantiate the contract
+
+If you want to test what you have so far, you can upload the contract using the [Contracts UI](https://paritytech.github.io/contracts-ui).
+
+To test the ERC-20 contract before adding new functions:
+
+1. Start the local contract node.
+
+1. Upload the `erc20.contract` file.
+
+1. Specify an initial supply of tokens for the `new` constructor.
+
+1. Instantiate the contract on the running local node.
+
+1. Select `totalSupply` as the Message to Send, then click **Read** to verify that the total supply of tokens is the same as the initial supply.
+
+1. Select `balanceOf` as the Message to Send.
+
+1. Select the `AccountId` of the account used to instantiate the contract, then click **Read**.
+    
+    If you select any other AccountId, then click **Read**, the balance is zero because all of the tokens are owned by the contract owner.
```

Comment:
```suggestion
    If you select any other `AccountId`, then click **Read**, the balance is zero because all of the tokens are owned by the contract owner.
```